\chapter{Introduction}\label{intro}

\section{Overview}

Cancer is a genetic disease that for many years leads the number of global deaths \citep{Bray2021TheWorldwide}. Cancer development, or carcinogenesis, occurs as a consequence of changes in the normal cells that stimulate cell proliferation \citep{Weinberg1996HowArises}. This often forms tumours that damage the healthy part of the local tissues by invasion, increased infection, interference with the organ function and so on \citep{Tobias2014CancerManagement}. The process of carcinogenesis is complicated by the diversity in the forms cancer can possibly takes \citep{weinberg2013biology}. In particular, there are roughly 100 cancers making up the major types: \gls{carcinoma}, \gls{lymphoma}, \gls{leukemia}, \gls{sarcoma} and \gls{neuroectodermal_tumour}. Cancer treatment relies on our understanding of the specific cancer sample. For instance, trastuzumab, which targets the HER2 protein, is known to deliver good response in HER2-positive, but poor response in HER2-negative breast cancers \citep{Kreutzfeldt2020TheTherapies}. Another key to cancer treatment is early diagnosis \citep{Hawkes2019CancerDiagnosis}. For example, the 5-year survival rate for prostate cancer is 100\% \textit{v.s.} 48\% if intervened at stage 1 and 4, respectively. 

Genetically, the process of \gls{carcinogenesis} is operated and characterised by mutations \citep{Stratton2009}. Whether a mutation occurs is in turn determined by whether mutagens can form a \gls{lesion} in the DNA and whether the repair systems can correctly fix the lesion \citep{Chatterjee2017MechanismsMutagenesis}. While every cancer has a different set of mutations, certain mutation patterns have been associated with certain cancer types \citep{Alexandrov2013,Polak2015,Campbell2020}. Since all cancers originate from a normal cell \citep{Hanahan2011HallmarksGeneration}, different patterns of cancer \gls{mutagenesis} are likely influenced by the mutation tendency in the original cell type. During its development, the phenotype of a cancer sample could diverge from that of the original cell, but virtually all mutations prior to the divergence remain in the cancer genome (Figure \ref{fig:drivers_demo}). These divergence events can be caused by driver mutations, \textit{i.e.} mutations that promote cancer cell proliferation \citep{Pon2015}. An average cancer sample consists of only about 4-5 drivers, the rest are passenger mutations, which have neutral effects on the cancer progress \citep{Campbell2020}. Together, the whole history of mutations in a cancer sample helps characterise its mutation profile. 

\input{figures/1_drivers_demo}

\subsection{Significance}
The mutation profile reflects the mechanism of mutagenesis, which offers a great opportunity to study cancer and to develop a cancer classifier that relies purely on mutation data. This project seeks to analyse the invaluable information extracted from the cancer mutation profile to elucidate this mechanism. Currently, precision medicine research focuses on targeting driver genes \citep{Mukherjee2019Genomics-GuidedCancer}. For this reason, an enhanced understanding of mutations independently of gene could perhaps expand the space for cancer therapy. \citet{Chowdhury2018PresenceNucleotides} indeed proposed to introduce the CG motif into the genome of dividing cancer cells because this motif is very vulnerable to mutations - the rationale was to neutralise the mutation rate in important regions of the genome. Furthermore, I seek to exploit this information to predict cancer types by training a \gls{classifier} that only relies on genomic sequencing data. In the long term, such a classifier could be an additional diagnostic tool to existing clinical approaches such as cytology or biopsy \citep{Stone1995Biopsy:Pitfalls}. In this era of next generation sequencing, liquid biopsies are gaining interest as a powerful non-invasive method for early cancer diagnosis because they involve screening for circulating tumour DNA in the blood rather than obtaining samples from a suspected local tissue \citep{Chen2019Next-generationDetection}. Developing a genome-based classifier model means that liquid biopsies could inform not only whether a cancer is present, but also where and what cancer is occurring at an early stage.  
\newpage
\subsection{Scope}
In this project, I studied two aspects of a cancer mutation profile: (1) where mutations tend to be found in the genome (hereafter the Genomic Location Effect, GLE), and (2) what base changes tend to be found in which genomic sequence context (hereafter the Sequence Context Effect, SCE). The scope of this project is limited to point mutations, which are the most available type of mutations in cancer \citep{Alexandrov2020}. Additionally, since driver and passenger mutations have different mutation rates, they should be considered separately. In this project, I focused on passenger mutations because of their abundance \citep{McFarland2014Tug-of-warProcesses}. Finally, I only investigated \glspl{sommut} as opposed to \glspl{germline_mut}, acknowledging that germline mutations could themselves be a risk factor of cancer. The reason for this is that germline mutations are present in effectively all the cells of a person, hence they are not the direct consequence of the mutagenic environment (Figure \ref{fig:drivers_demo}). 

Section \ref{intro:gle} and \ref{intro:sce} of this chapter review what is known about GLE and SCE, respectively. Section \ref{intro:ml} briefly introduces the computational approaches used to study GLE and SCE, as well as how they can be used to train a machine learning classifier. Section \ref{intro:aims} summarises the aims and hypotheses of the project. Section \ref{intro:findings} outlines the key findings.

\section{Genomic location effect (GLE)}
\label{intro:gle}
Certain regions of the genome are more prone to mutations than others, with the locations of these regions in the genome varying in different cell and cancer types \citep{Polak2015, Jiao2020}. This is believed to result from the fact that different cells have different chromatin structures \citep{Abascal2020ExpandedGenomes}, which determine how hidden DNA is in the chromatin complex across the genome. The accessibility of DNA to transcription, mutagens and repair systems can be measured by Dnase I hypersensitivity \citep[DHS;][]{Liu2019AApplications}. Essentially, DHS measures the accessibility of a genomic region by exposing that region to Dnase, which cleaves the DNA. The more severely cleaved, the more accessible the region is. Highly accessible DNA means open chromatin status, and vice versa. Interestingly, it has been reported that open chromatin regions are less likely to harbour mutations than closed regions \citep{Polak2015,Prendergast2007ChromatinGenome}. Given that \glspl{mutagen} interact with repairs to determine whether mutations occur \citep{Ripley2001Mutation}, and that DNA is less accessible to both in closed regions, this suggests that the repair effect is generally stronger \citep[Figure \ref{fig:chromatin_demo};][]{Teng1997ExcisionSequences, Morse2002PhotoreactivationCerevisiae}. That said, the questions remain how strong chromatin structure is as a determinant of the genomic location of mutations. In addition, it is equally intriguing to examine whether GLE, as a result of such determinants, can act as a distinct characteristic of the cancer mutation profile. 

\input{figures/1_chromatin_demo}
\newpage
To represent genomic location data, the convention is to bin the genome, here after the bin method \citep{Kubler2019, Salvadores2019PassengerTumors, Chalmers2017AnalysisBurden, Salvadores2020MatchingPatterns}. The bin method divides the genome into consecutive segments of 1 Mbp in length, then counts the number of mutations in these segments. However, the 1 Mbp size is picked by human, which imposes arbitrary boundaries to the genome. This leads to an unstable representation that is distorted when only slightly shifting the start of the bin boundaries (Figure \ref{fig:mutdistribution_demo}). To counteract this, my project experimented with smoothing GLE by estimating the \gls{density} at a genomic location based on the amount of data adjacent to it\footnote{details in }. The smooth representation assumes no rigid boundaries to the genome, and is thus less sensitive to the above pitfall. 
\input{figures/1_mutdistribution_demo}

\newpage
\section{Sequence context effect (SCE)}
\label{intro:sce}

Each cancer develops under the influence of different mutagenic processes, giving rise to diverse mutation compositions. These processes include, but are not limited to UV light \citep[known to drive skin melanoma;][]{Mohania2017}, the intrinsic cellular APOBEC deaminase activity \citep[\textit{e.g.} in B cells;][]{Kuppers2005MechanismsPathogenesis} and defective DNA repairs \citep[\textit{e.g.} mutated \textit{BRCA} genes in breast cancer;][]{Navasardyan2021YY1TNBC}. \citet{Alexandrov2013, Alexandrov2020} showed that some processes were associated with certain mutation signatures. For instance, the signature SBS4, where there is an excess of C$\rightarrow$A mutations in the context of C[C$\rightarrow$A]A and C[C$\rightarrow$A]C over any other mutations, was only detected in tobacco smoke linked cancers such as liver hepatocellular carcinoma, lung adenocarcinoma and lung squamous cell carcinoma. As such, similar to \gls{gle}, SCE is also an important characteristic of the cancer mutation profile. 

In inspecting SCE, it is important to note that mutations are usually closely linked with the \glspl{base} next to them \citep{Zhu2017}. A well known example is that the C$\rightarrow$T mutations tend to occur in the [C$\rightarrow$T]G context. One possible explanation is that proteins, be they repairs or mutagens, interact with DNA with high specificity (Figure \ref{fig:motif_demo}). There are two potential flaws to the standard way of representing SCE. First, while it is common practice to analyse mutation compositions in the 3-mer (3 base) context, evidence showed that bases beyond 3-mer could also influence the likelihood of mutations occurring \citep{Zhu2017,Zhu2020}. Second, base changes are generally assumed to be symmetric, meaning G$\rightarrow$A is assumed to always occur when C$\rightarrow$T occurs on the complementary strand \citep[Figure \ref{fig:motif_symmetric_demo};][]{Alexandrov2013, Jiao2020}; but analyses in skin melanoma showed otherwise \citep{Zhu2017}. Accordingly, I seek to explore the information content available in different sequence context sizes, particularly outer positions to 3-mers; as well as the effect of strand symmetry/asymmetry in representing SCE.

\input{figures/1_motif_demo}
\input{figures/1_motif_symmetric_demo}

\section{Computational approaches to study GLE and SCE}
\label{intro:ml}

The release of the \gls{pcawg} project \citep{Campbell2020}, with whole genomic sequencing data for 2605 primary tumours of 38 cancer types, has made studying cancer genomics and developing cancer classification models on a large scale feasible. Nevertheless, analysing developing methods based on the PCAWG mutation data requires satisfactory computational tools for data analysis. To develop these tools, it is necessary to understand the nature of the data, which I approached on two levels, analysis on whole disease scale and classification on individual scale. First, data was investigated on the whole disease level (\gls{gle} in Chapter \ref{gle}; \gls{sce} in Chapter \ref{sce}). This means that for each cancer, mutations from all donors of that cancer were considered as a whole. This makes understanding the mutagenesis process easier as it magnifies the signals in the data if they are ``real''. Second, Chapter \ref{ml} trials different measures and data representations to train the classifiers on the individual donor level. The idea is that the more accurate the classifier, the better the assumptions imposed by its data representations could capture the nature of the data. The long term goal is to be able to apply the model to unseen data, such as the tumour genome of a new cancer patient. The analyses on two levels are complementary, in that they could be used to verify each other. Whereas whole disease analysis explores how and why a potential factor might be important, individual scale is a direct measure of how informative that factor is.  

\section{Aims and questions}
\label{intro:aims}
\begin{enumerate}
    \item To evaluate the influence of chromatin structure on genomic location effect (GLE) and the degree at which GLE can discriminate cancers.
    \begin{itemize}
        \item Are mutations significantly biased towards closed or open regions?
        \item Is the bias uniform for all cancers?
        \item Is the proposed smoothing representation better at extracting information than the conventional bin approach?
    \end{itemize}
    \item To understand how sequence context effect (SCE) is characteristic of cancers
    \begin{itemize}
        \item Can the base substitutions without the flanking bases discriminate cancers?
        \item Are base substitutions strand symmetric
        \item Are the flanking bases informative? Is there any advantage in studying bases at outer positions such as position -2 and +2?
    \end{itemize}
    \item To train a classifier that accurately predict cancers for individual donors.
    \begin{itemize}
        \item Do the proposed methods for GLE and SCE outperform those normally used?
        \item Does the representations used by the best classifier match those identified by whole disease analyses?
        \item Does combining SCE and GLE results in further accuracy improvement than each factor alone.
    \end{itemize}
\end{enumerate}

\section{Key findings}
\label{intro:findings}
Overall, the project shows that both GLE and SCE are important characteristics of a cancer \gls{mut_profile}, manifesting in three aspects. First, mutations generally tend to occur in closed chromatin regions, but the degree of bias varies for different cancers. This suggests that chromatin structure is has an important influence on GLE, but it is not the single determinant of GLE. Regardless of the driving forces, GLE was found to be characteristic of cancers, particularly when the smoothing representation was used. Second, both components of SCE, the base changes and their sequence contexts are characteristic of cancers. Not surprisingly, information is more abundant closer to the mutations (3-mer), but the outer bases are also very informative. Additionally, there is more information in \glspl{transition} than in \glspl{transversion} for both base substitutions and their flanking bases, with evidence of strand symmetry. When recruited to train classifiers, as expected, smoothing GLE in any form provided a higher predictive power than binning it. Regarding SCE, incorporating the immediate neighbours within the 3-mer neighbourhood improved accuracy over the base changes alone. In addition, while information was detected in the outer bases of the 5-mer sequence context, incorporating this information into the classifier appeared to be quite challenging. Finally, even though GLE and SCE were expected to be separate sources of information, SCE were the predominating contributor of the combined classifier. To be exact, there was no improvement in predictive power in the combined classifier compared to the SCE classifier when using the proposed representations, and very little improvement when using the conventional representations. With the findings of my project, I hope the understanding of cancer mutagenesis could help guiding further research in therapeutics. By pointing out the flaw in the standard representation of data, we are now hopefully one step closer to an applicable mutation-based cancer classifier. 

